lib/chef-dk/command/verify.rb

```diff
@@ -141,6 +147,35 @@ def components
           # ------------
           failures = []
 
+          # fail the verify if we have more than one version of chef-provisioning or any of its drivers.
```

Comment:
Doing it for chef-dk as a whole is also going to get tricky - we'd only want to do it for the gems "we control".  2 versions of rspec or cucumber or nokogiri might be inevitable.